providers/fips/fipsprov.c

```diff
@@ -518,7 +518,22 @@ static const OSSL_DISPATCH intern_dispatch_table[] = {
     { 0, NULL }
 };
 
+/*
+ * On VMS, the provider init function name is expected to be uppercase,
+ * see the pragmas in <openssl/core.h>.  Let's do the same with this
+ * internal name.  This is how symbol names are treated by default
+ * by the compiler if nothing else is said, but since this is part
+ * of libfips, and we build our libraries with mixed case symbol names,
+ * we must switch back to this default explicitly here.
+ */
+#ifdef __VMS
+# pragma names save
+# pragma names uppercase,truncated
+#endif
 OSSL_provider_init_fn OSSL_provider_init_int;
+#ifdef __VMS
```

Comment:
Should this be _after_ the function or does VMS remember changing the name after the restore?

Reply:
Yes.  What this pragma does is basically to set a name treatment attribute on that *name*.  You'll notice that \<openssl/core.h\> does exactly the same for `OSSL_provider_init`.